kiva/tests/test_agg_drawing.py

```diff
@@ -46,3 +46,27 @@ def test_ipython_repr_png(self):
         with open(filename, 'wb') as fp:
             fp.write(stream)
         self.assertImageSavedWithContent(filename)
+
+    def test_rasterizer_cell_overflow(self):
+        def genpoints(num):
+            arr = np.empty((num, 2))
+            arr[::2, 0] = np.linspace(0, 99, num=arr[::2, 0].shape[0])
+            arr[::2, 1] = 0.0
+            arr[1::2, 0] = np.linspace(1, 100, num=arr[1::2, 0].shape[0])
+            arr[1::2, 1] = 100.0
+            return arr
+
+        # This many points definitely generates more than 2^22 cells
+        count = 10000
+        points = genpoints(count)
+        self.gc.lines(points)
+        with self.assertRaises(OverflowError):
+            self.gc.stroke_path()
+
+        self.gc.begin_path()
```

Comment:
Not sure if we want to do this, but without this `begin_path` the next draw will trigger the exception again because throwing an exception caused the path to not be cleared by the draw.

Reply:
The issue could be fixed by adding a path reset to the exception handler.

Reply:
Done